### DISCUSSION TITLE
Favourites transposed...

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#2428] [General]
I opened Sumatra and one of the PDF's favourites had become mangled by becoming set against another PDF that hadn't been favourited, yet leaving 2 favourites remaining; the PDF also didn't exist and was perhaps a corruption for something else, but was adjacent (prior to) the one that had "lost" its favourites. I did repair it manually in the settings file.
Well I can't detail any more than that but perhaps when looking at the code that saves the favourites (perhaps when you add a favourite) keep an eye out for the kind of logic error that might cause this to happen...
I've not had it happen before but I upgraded to 3.4.14300 (64-bit) very recently.
David

**Comment by GitHubRulesOK** [comment]
Favorites are per filename so if another file gets renamed externally it will cause problems for any app that uses that filename or is on a drive that is plugged out and in, thus the file has a replacement due to paths.
```
FilePath = C:\project\my.pdf
		Favorites [
			[
				Name = Fred
				PageNo = 1
			]
```
